src/ucp/wireup/wireup_cm.c

```diff
@@ -192,35 +192,33 @@ static ssize_t ucp_cm_client_priv_pack_cb(void *arg, const char *dev_name,
 free_addr:
     ucs_free(ucp_addr);
 out:
-    if (status != UCS_OK) {
+    if (status == UCS_OK) {
+        ep->flags |= UCP_EP_FLAG_LOCAL_CONNECTED;
```

Comment:
isn't this too early to set this flag? should happen after getting connect ack from server?
is it ok to send disconnect after this? because this flag decides whether to send disconnect or not..

Reply:
ucp_wireup_ep + flush protects from disconnecting before the EP is remotely connected.